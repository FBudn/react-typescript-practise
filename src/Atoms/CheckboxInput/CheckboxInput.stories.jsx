import React from "react";
import CheckboxInput from "./CheckboxInput";

export default {
  title: "Atoms/CheckboxInput",
  component: CheckboxInput,
  argTypes: { ref: { action: "I'm forwarding a reference" } },
};

const Template = (args) => <CheckboxInput {...args} />;

export const StyledChecbox = Template.bind({});
StyledChecbox.args = {
  id: 1,
};
