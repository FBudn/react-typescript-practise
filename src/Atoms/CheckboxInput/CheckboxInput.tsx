import React, { forwardRef } from "react";
import { CheckboxInputContainer } from "./CheckboxInput.style";

export interface CheckboxInputProps {
  id?: string;
  testId?: string;
}

export type CheckboxRef = HTMLInputElement;
const CheckboxInput = forwardRef<CheckboxRef, CheckboxInputProps>(
  ({ id, testId }, ref) => (
    <CheckboxInputContainer data-testid={testId} id={id} ref={ref} />
  ),
);

export default CheckboxInput;

/* 
Używam checkboxa tylko raz, nieużywam tu id akurat - może do usunięcia by przeszło jak jest już testoweId?
Ogólnie wszędzie może bym dodał inne wersje stylów itp, ale w sumie pod konkretny projekt nie wiem czy to 
konieczne
Nazewnictwo chyba ok


podkreślnik w funkcji tu: można używać do nazw zmiennych, jako ukryty nieużywany parament 

const CheckboxInput = forwardRef(function CheckboxInput(
  props: CheckboxInputProps,
  ref: CheckboxRef,
) {
  return <CheckboxInputStyle ref={ref} />;
}); 

export interface CheckboxInputProps {
  // onChangeCheckbox: (event: React.ChangeEvent<HTMLInputElement>) => void;
  // tu się nie definiuje refa , bez propsów ogarnąć dzałanie bo tu nie mam propsów i tak
  id?: string;
}
export type CheckboxRef = HTMLInputElement;
const CheckboxInput = forwardRef<CheckboxRef, CheckboxInputProps>(
  ({ id }, ref) => <CheckboxInputStyle id={id} ref={ref} />,
);

onChange={onChangeCheckbox}
const TestOrganism = forwardRef(function TestOrganism(props: any, ref) {
  const { label, onChangeInput, ...otherProps } = props;
  return (
    <InputTest>
      {label}
      <InputAtom {...otherProps} ref={ref} onChange={onChangeInput} />
    </InputTest>
  );
});

onChange na poziomie atomu z nazwą też onChangeCośtam

import { forwardRef } from "react";
import { CheckboxInputStyle } from "./CheckboxInputStyle";

const CheckboxInput = forwardRef(function CheckboxInput(props, ref) {
  const  onChangeCheckbox  = props;

  return (
  <CheckboxInputStyle onChange={onChangeCheckbox} ref={ref} />;
);
});

export default CheckboxInput;
*/
