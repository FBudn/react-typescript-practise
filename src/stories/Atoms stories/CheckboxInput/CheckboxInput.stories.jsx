import React from "react";
import CheckboxInput from "../../../Atoms/CheckboxInput/CheckboxInput";

export default {
  title: "Atoms/CheckboxInput",
  component: CheckboxInput,
  argTypes: { ref: { action: "I'm forwarding a reference" } },
};

const Template = (args) => <CheckboxInput {...args} />;

export const StyledChecbox = Template.bind({});
StyledChecbox.args = {
  id: 1,
  onChange: `Forwarding reference`,
};
