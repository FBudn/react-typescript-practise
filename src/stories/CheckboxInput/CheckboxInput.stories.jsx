/* eslint-disable react/jsx-props-no-spreading */
import React, { useRef } from "react";
import CheckboxInput from "../../Atoms/CheckboxInput/CheckboxInput";

export default {
  title: "Atoms/CheckboxInput",
  component: CheckboxInput,
};

const Template = (args) => <CheckboxInput {...args} />;

export const StyledChecbox = Template.bind({});
StyledChecbox.args = {
  id: 1,
  useRef,
  onChange: `Forwarding reference`,
};
