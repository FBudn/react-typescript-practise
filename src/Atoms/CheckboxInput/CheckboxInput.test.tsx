/* eslint-disable react/react-in-jsx-scope */
import { render, screen } from "@testing-library/react";
import CheckboxInput from "./CheckboxInput";

test(`Should render CheckboxInput Component`, () => {
  render(<CheckboxInput />);
  const checkboxInputElement = screen.getByTestId(`CheckboxInput-1`);
  expect(checkboxInputElement).toBeInTheDocument();
});
