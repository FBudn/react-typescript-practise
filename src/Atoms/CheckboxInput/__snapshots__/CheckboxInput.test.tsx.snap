// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match snapshot and have styles 1`] = `
.c0 {
  max-width: 35px;
  min-width: 20px;
  max-height: 35px;
  min-height: 20px;
  margin: 5px 0px 0px 0px;
  cursor: pointer;
}

<input
  className="c0"
  data-testid="CheckboxInput-1"
  type="checkbox"
/>
`;
